base/daemon/BaseDaemon.cpp

```diff
@@ -389,20 +390,16 @@ static void sanitizerDeathCallback()
 
     const StackTrace stack_trace;
 
-    StringRef query_id = DB::CurrentThread::getQueryId();
-    query_id.size = std::min(query_id.size, max_query_id_size);
-
     int sig = SignalListener::SanitizerTrap;
     DB::writeBinary(sig, out);
     DB::writePODBinary(stack_trace, out);
     DB::writeBinary(UInt32(getThreadId()), out);
-    DB::writeStringBinary(query_id, out);
     DB::writePODBinary(DB::current_thread, out);
 
     out.next();
 
     /// The time that is usually enough for separate thread to print info into log.
-    sleepForSeconds(10);
+    sleepForSeconds(20);
```

Comment:
What is the reason for doubling this?

Reply:
> What is the reason for doubling this?

Same to `signalHandler`, which may be not necessary  at all.

https://github.com/ClickHouse/ClickHouse/blob/9e91a9dfd1dae8072d9d2132a4b3e4bbb70e1c1d/base/daemon/BaseDaemon.cpp#L150-L152

Reply:
Ok, I see, it had been increased in #19317